### ISSUE TITLE
Hebrew calendar and Jewish holidays, 

### ISSUE THREAD

**Issue by rosennej** [MenoData/Time4J#872]
1. I suggest adding the Jewish holidays to the Hebrew calendar. I know it is quite complicated. I could help with advice and consultation with calendar experts.
2. I disagree with the statement "The civil Hebrew day starts at 18:00 o'clock on the previous day." The banking day in Israel starts at 18:30 on weekdays, 14:00 on Fridays, unless they are holidays. Sunday is a weekday. The banking day is legally the business day unless specified differently. Before 2006 the banking day started at 15:00. Where the Hebrew calendar is used for religious purposes is is always sunset. Same outside of Israel, as far as I know.

**Comment by MenoData** [comment]
about 2: See the other new issue for better documentation. I agree that the documentation should avoid the term "civil day". It is not related to banking days in Israel.

about 1: Thank you very much for your offer. I know at least following sources for holiday handling:

- https://en.wikipedia.org/wiki/Jewish_and_Israeli_holidays_2000–2050
- the book "Calendrical calculations" from Dershowitz/Reingold

Here the general question arises if we might choose a new module for holidays which depends on Time4J-base. Main reason is that it might require more frequent updating - not so much for Jewish holidays but for holidays in any countries. The new module would be like a database which is open for frequent additions and changes. It would manage different types and validities of holidays in countries or regions. We also have the interface `net.time4j.range.HolidayModel` which should be implemented by Jewish holidays.

However, contributors for such an updatable holiday module are very welcome. I am not sure if there will be enough contributors willing to contribute changes.

**Comment by rosennej** [comment]
What was the intended meaning of "civil day"?

**Comment by MenoData** [comment]
It was just my faulty oversimplification. "civil" = "non religious" with the consequence that the exact time of evening is not important. Anyway, if a user does not pass any start-of-day-parameter then Time4J can only guess, and 18:00 is a (rough) approximation which works best near the equator. The documentation should explain this behaviour explicitly using the word "approximation".

By the way, if we go down to minutes or even seconds (a little bit questionable in context of possible accuracy of astronomic calculations) then some religious Jews even seem to make strict differences betwen various sunset- and nightfall-definitions, see for example: https://www.yeshiva.co/ask/?id=7469

**Comment by rosennej** [comment]
The same applies to Hijri, AFAIK. For example, what to do you when you are in Iceland in the summer or the winter?

**Comment by MenoData** [comment]
Users should pass a `StartOfDay`-Parameter with the appropriate geographic location for achieving best results, equal if Hijri or Hebrew calendar.

**Comment by MenoData** [comment]
Oh, and when applied to polar regions which don't observe sunset at all in summer then only conventions like setting the geographic latitude to 60° N might help (as some religous authorities have suggested as compromise). This can be done, too, by passing an appropriate `StartOfDay`-definition. The user has all freedom.
